Flaky test `CompleteUserTaskTest.initClientAndInstances`

**Summary**

> Try to answer the following as best as possible

- How often does the test fail?
- Does it block your work?
- Do we suspect that it is a real failure? No, it appears to be a wrong port selection

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

```
Caused by: org.springframework.boot.web.server.PortInUseException: Port 1840 is already in use
```

<details><summary>Example assertion failure</summary>
<pre>
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    CompleteUserTaskTest.initClientAndInstances:52 [Expected a created user task] 
Expecting actual:
  -1L
to be greater than:
  0L

Error:    CompleteUserTaskTest.initClientAndInstances:52 [Expected a created user task] 
Expecting actual:
  -1L
to be greater than:
  0L

Error:  Errors: 
Error:  io.camunda.zeebe.it.client.command.CompleteUserTaskTest.null
Error:    Run 1: CompleteUserTaskTest » Completion org.springframework.context.ApplicationContextException: Failed to start bean 'webServerStartStop'
Error:    Run 2: CompleteUserTaskTest » Completion org.springframework.context.ApplicationContextException: Failed to start bean 'webServerStartStop'
Error:    Run 3: CompleteUserTaskTest » Completion org.springframework.context.ApplicationContextException: Failed to start bean 'webServerStartStop'
[INFO] 
[INFO] 
Error:  Tests run: 458, Failures: 2, Errors: 1, Skipped: 0
</pre>
</details>

**Hypotheses**

> List any hypotheses if you have one; can be ommitted

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

Full test results: [Google Drive](https://drive.google.com/file/d/1E4Sul7m3v7coJWxhtCUKXpDLToWc_-G9/view?usp=sharing)

<details><summary>Logs</summary>
<p>

```
[INFO] Running io.camunda.zeebe.it.client.command.CompleteUserTaskTest
Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.587 s <<< FAILURE! -- in io.camunda.zeebe.it.client.command.CompleteUserTaskTest
Error:  io.camunda.zeebe.it.client.command.CompleteUserTaskTest -- Time elapsed: 0.587 s <<< ERROR!
java.util.concurrent.CompletionException: org.springframework.context.ApplicationContextException: Failed to start bean 'webServerStartStop'
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1807)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
Caused by: org.springframework.context.ApplicationContextException: Failed to start bean 'webServerStartStop'
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:291)
	at org.springframework.context.support.DefaultLifecycleProcessor$LifecycleGroup.start(DefaultLifecycleProcessor.java:471)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at org.springframework.context.support.DefaultLifecycleProcessor.startBeans(DefaultLifecycleProcessor.java:260)
	at org.springframework.context.support.DefaultLifecycleProcessor.onRefresh(DefaultLifecycleProcessor.java:205)
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:981)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:627)
	at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:66)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:456)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:334)
	at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:149)
	at io.camunda.zeebe.qa.util.cluster.TestSpringApplication.start(TestSpringApplication.java:71)
	at io.camunda.zeebe.qa.util.cluster.TestStandaloneGateway.start(TestStandaloneGateway.java:22)
	at io.camunda.zeebe.qa.util.cluster.TestSpringApplication.start(TestSpringApplication.java:25)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
	... 6 more
Caused by: org.springframework.boot.web.server.PortInUseException: Port 1840 is already in use
	at org.springframework.boot.web.embedded.netty.NettyWebServer.lambda$start$2(NettyWebServer.java:135)
	at org.springframework.boot.web.server.PortInUseException.ifCausedBy(PortInUseException.java:103)
	at org.springframework.boot.web.embedded.netty.NettyWebServer.start(NettyWebServer.java:133)
	at org.springframework.boot.web.reactive.context.WebServerManager.start(WebServerManager.java:55)
	at org.springframework.boot.web.reactive.context.WebServerStartStopLifecycle.start(WebServerStartStopLifecycle.java:41)
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:288)
	... 21 more
Caused by: reactor.netty.ChannelBindException: Failed to bind on [0.0.0.0:1840]
	Suppressed: java.lang.Exception: #block terminated with an error
		at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:146)
		at reactor.core.publisher.Mono.block(Mono.java:1807)
		at reactor.netty.transport.ServerTransport.bindNow(ServerTransport.java:149)
		at reactor.netty.transport.ServerTransport.bindNow(ServerTransport.java:134)
		at org.springframework.boot.web.embedded.netty.NettyWebServer.startHttpServer(NettyWebServer.java:185)
		at org.springframework.boot.web.embedded.netty.NettyWebServer.start(NettyWebServer.java:130)
		at org.springframework.boot.web.reactive.context.WebServerManager.start(WebServerManager.java:55)
		at org.springframework.boot.web.reactive.context.WebServerStartStopLifecycle.start(WebServerStartStopLifecycle.java:41)
		at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:288)
		at org.springframework.context.support.DefaultLifecycleProcessor$LifecycleGroup.start(DefaultLifecycleProcessor.java:471)
		at java.base/java.lang.Iterable.forEach(Iterable.java:75)
		at org.springframework.context.support.DefaultLifecycleProcessor.startBeans(DefaultLifecycleProcessor.java:260)
		at org.springframework.context.support.DefaultLifecycleProcessor.onRefresh(DefaultLifecycleProcessor.java:205)
		at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:981)
		at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:627)
		at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:66)
		at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754)
		at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:456)
		at org.springframework.boot.SpringApplication.run(SpringApplication.java:334)
		at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:149)
		at io.camunda.zeebe.qa.util.cluster.TestSpringApplication.start(TestSpringApplication.java:71)
		at io.camunda.zeebe.qa.util.cluster.TestStandaloneGateway.start(TestStandaloneGateway.java:22)
		at io.camunda.zeebe.qa.util.cluster.TestSpringApplication.start(TestSpringApplication.java:25)
		at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
		at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796)
		at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
		at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
		at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
		at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
		at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
```

</p>
</details>

--- a/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/CompleteUserTaskTest.java
+++ b/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/CompleteUserTaskTest.java
@@ -30,8 +30,7 @@ import org.junit.jupiter.api.Test;
 class CompleteUserTaskTest {
 
   @TestZeebe
-  private static final TestStandaloneBroker ZEEBE =
-      new TestStandaloneBroker().withRecordingExporter(true);
+  private final TestStandaloneBroker zeebe = new TestStandaloneBroker().withRecordingExporter(true);
 
   @AutoCloseResource private ZeebeClient client;
 
@@ -39,7 +38,7 @@ class CompleteUserTaskTest {
 
   @BeforeEach
   void initClientAndInstances() {
-    client = ZEEBE.newClientBuilder().defaultRequestTimeout(Duration.ofSeconds(15)).build();
+    client = zeebe.newClientBuilder().defaultRequestTimeout(Duration.ofSeconds(15)).build();
     final ZeebeResourcesHelper resourcesHelper = new ZeebeResourcesHelper(client);
     userTaskKey = resourcesHelper.createSingleUserTask();
   }

--- a/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/UnassignUserTaskTest.java
+++ b/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/UnassignUserTaskTest.java
@@ -26,8 +26,7 @@ import org.junit.jupiter.api.Test;
 class UnassignUserTaskTest {
 
   @TestZeebe
-  private static final TestStandaloneBroker ZEEBE =
-      new TestStandaloneBroker().withRecordingExporter(true);
+  private final TestStandaloneBroker zeebe = new TestStandaloneBroker().withRecordingExporter(true);
 
   @AutoCloseResource private ZeebeClient client;
 
@@ -35,7 +34,7 @@ class UnassignUserTaskTest {
 
   @BeforeEach
   void initClientAndInstances() {
-    client = ZEEBE.newClientBuilder().defaultRequestTimeout(Duration.ofSeconds(15)).build();
+    client = zeebe.newClientBuilder().defaultRequestTimeout(Duration.ofSeconds(15)).build();
     final ZeebeResourcesHelper resourcesHelper = new ZeebeResourcesHelper(client);
     userTaskKey = resourcesHelper.createSingleUserTask();
   }

--- a/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/UpdateUserTaskTest.java
+++ b/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/UpdateUserTaskTest.java
@@ -34,8 +34,7 @@ class UpdateUserTaskTest {
       ZonedDateTime.of(2023, 11, 11, 11, 11, 11, 11, ZoneId.of("UTC")).toString();
 
   @TestZeebe
-  private static final TestStandaloneBroker ZEEBE =
-      new TestStandaloneBroker().withRecordingExporter(true);
+  private final TestStandaloneBroker zeebe = new TestStandaloneBroker().withRecordingExporter(true);
 
   @AutoCloseResource private ZeebeClient client;
 
@@ -43,7 +42,7 @@ class UpdateUserTaskTest {
 
   @BeforeEach
   void initClientAndInstances() {
-    client = ZEEBE.newClientBuilder().defaultRequestTimeout(Duration.ofSeconds(15)).build();
+    client = zeebe.newClientBuilder().defaultRequestTimeout(Duration.ofSeconds(15)).build();
     final ZeebeResourcesHelper resourcesHelper = new ZeebeResourcesHelper(client);
     userTaskKey = resourcesHelper.createSingleUserTask();
   }

